x-pack/platform/plugins/shared/search_connectors/server/lib/indices/create_index.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { MappingKeywordProperty, MappingTextProperty } from '@elastic/elasticsearch/lib/api/types';
+import { IScopedClusterClient } from '@kbn/core/server';
+
+import { textAnalysisSettings } from './text_analysis';
+
+const prefixMapping: MappingTextProperty = {
+  analyzer: 'i_prefix',
+  index_options: 'docs',
+  search_analyzer: 'q_prefix',
+  type: 'text',
+};
+
+const delimiterMapping: MappingTextProperty = {
+  analyzer: 'iq_text_delimiter',
+  index_options: 'freqs',
+  type: 'text',
+};
+
+const joinedMapping: MappingTextProperty = {
+  analyzer: 'i_text_bigram',
+  index_options: 'freqs',
+  search_analyzer: 'q_text_bigram',
+  type: 'text',
+};
+
+const enumMapping: MappingKeywordProperty = {
+  ignore_above: 2048,
+  type: 'keyword',
+};
+
+const stemMapping: MappingTextProperty = {
+  analyzer: 'iq_text_stem',
+  type: 'text',
+};
+
+const defaultMappings = {
+  dynamic: true,
+  dynamic_templates: [
+    {
+      all_text_fields: {
+        mapping: {
+          analyzer: 'iq_text_base',
+          fields: {
+            delimiter: delimiterMapping,
+            enum: enumMapping,
+            joined: joinedMapping,
+            prefix: prefixMapping,
+            stem: stemMapping,
+          },
+        },
+        match_mapping_type: 'string',
+      },
+    },
+  ],
+};
+
+export const createIndex = async (
+  client: IScopedClusterClient,
+  indexName: string,
+  language: string | undefined | null,
+  applyMappings: boolean
+) => {
+  return await client.asCurrentUser.indices.create({
+    index: indexName,
+    mappings: applyMappings ? defaultMappings : {},
+    settings: {
+      ...textAnalysisSettings(language ?? undefined),
```

Comment:
I don't think we want this anymore. This shouldn't be used anymore in enterprise_search it just hasn't been removed yet.

Reply:
It is used by Elastic managed connector in add connector https://github.com/elastic/kibana/blob/b9264e57f92a2b9ce1e25ce3f791282b36a92337/x-pack/platform/plugins/shared/search_connectors/server/lib/connectors/add_connector.ts#L59

Reply:
There should not be any Elastic managed connectors in 9.x or serverless. Those are only available in 8.x with an Enterprise Search node.

There is likely code in `enterprise_search` that didn't get removed but isn't actually used in 9.x. or at least it's not supposed to be used.

cc: @seanstory

Reply:
:/ ok this is still used by connectors, we missed that in the 9.0.0 readiness work.